index.bs

```diff
@@ -346,10 +346,47 @@ interface mixin DatagramTransport {
 
 ## Procedures ## {#datagrams-transport-procedures}
 
-The <dfn>readDatagrams</dfn> algorithm is given a |transport| as parameter. It
-is defined by running the following steps:
-1. TODO
-1. [=ReadableStream/enqueue=] |data| in |transport|.`[[IncomingDatagrams]]`.
+To <dfn>pullDatagrams</dfn>, given a {{WebTransport}} object |transport|, run these steps:
+1. Let |queue| be |transport|'s [=[[IncomingDatagramsQueue]]=].
+1. If |queue| is empty, then:
+  1. Assert: |transport|'s [=[[IncomingDatagramsPullPromise]]=] is null.
+  1. Set |transport|'s [=[[IncomingDatagramsPullPromise]]=] to a new promise.
+  1. Return |transport|'s [=[[IncomingDatagramsPullPromise]]=].
+1. Let |bytes| and |timestamp| be |queue|'s first element.
+1. Remove the first element from |queue|.
```

Comment:
The datagrams can still be stale. receiveDatagrams will only be run when new datagrams are received on a connection/session and if that is delayed the datagrams in the queue may be older than IncomingDatagramsExpirationDuration.

Or is it suggested to run receiveDatagrams whenever a datagram go stale as well?

Reply:
> Or is it suggested to run receiveDatagrams whenever a datagram go stale as well?

Yes.

Reply:
[sendDatagrams](https://w3c.github.io/webtransport/#senddatagrams) is specified in the same manner.